client/starwhale/api/_impl/data_store.py

```diff
@@ -891,6 +896,7 @@ def scan(
         keep_none: bool = False,
         end_inclusive: bool = False,
         revision: Optional[str] = None,
+        deep_copy: bool = False,
```

Comment:
how to use `deep_copy` in the wrapper interface?

Reply:
The caller chooses whether to copy the data or not because it knows whether it will be modified.
As the doc said ([ref](https://docs.python.org/3/library/copy.html)):
>  A deep copy constructs a new compound object and then, recursively, inserts copies into it of the objects found in the original
> `deep copy` copies everything it may copy too much, such as data which is intended to be shared between copies

Reply:
for standalone, the scan call chain: dataset wrapper `scan` ->  LocalDataStore `scan_tables` ->  MemoryTable `scan`. 

The `deep_copy` option is only for MemoryTable `scan`.

Reply:
Made deep copy true by default